### ISSUE TITLE
`torch.argmax` treats `axis` as an alias for `dim`, but this isn't documented.

### ISSUE THREAD

**Issue by akshatvishu** [pytorch/pytorch#103977]
### 📚 The doc issue

Actually was going through How [Dirichlet distribution](https://github.com/pytorch/pytorch/blob/678ce61cdb0e652ee4dfbb236319bd6acb98b851/torch/distributions/dirichlet.py#L85) is implemented but I found this at line 85 :
```python
mode[mask] = torch.nn.functional.one_hot(mode[mask].argmax(axis=-1), concentrationm1.shape[-1]).to(mode)
```
and when I checked the documents for [torch.argmax](https://pytorch.org/docs/stable/generated/torch.argmax.html), there is no mentioned that  `axis `  can be used in-place of `dim`.
```python
import torch

a = torch.randn(2,2)
b = torch.argmax(a, dim=1)
c = torch.argmax(a, axis=1)
print(b) # tensor([0, 1])
print(c) # tensor([0, 1])
```
I mostly curious that if this done deliberately  to maintain the same function signature as [numpy.argmax](https://numpy.org/doc/stable/reference/generated/numpy.argmax.html), if so what other function follow the same behavior  ?

### Suggest a potential alternative/fix

Should add a note that `axis` can be used interchangeably with `dim`
